src/components/posts/PostItem.js

```diff
@@ -41,16 +42,38 @@ function PostItem({ likes, content, avatar, id, author, date }) {
             <div className={classes.likeCount}>{likeCount}</div>
             <div className={classes.actions}>
               <div className={classes.like}>
-                <LikeButton
+                <ButtonAction
+                  buttonTitle="like"
                   id={id}
+                  buttonFunction={LikeHandler}
                   setLikeCount={setLikeCount}
                   likeCount={likeCount}
                 />
               </div>
               <button onClick={editHandler}>edit</button>
-              <DeleteButton id={id} setPostDelete={setPostDelete} />
+              <button onClick={deleteHandler}>delete</button>
             </div>
           </Post>
+          {showDeleteAlert && (
+            <>
+              <Post>
+                <div className={classes.modal}>
```

Comment:
This confirmation dialogue is doing something different enough from the current component that it should probably be a new component. The general rule of thumb is single responsibility, every function, component or sufficiently modular aspect of the code should be in a separate section or file and I think this is doing something that is separable.